collect_app/src/main/java/org/odk/collect/android/external/AndroidShortcutsActivity.kt

```diff
@@ -45,7 +45,11 @@ class AndroidShortcutsActivity : AppCompatActivity() {
         super.onCreate(bundle)
         DaggerUtils.getComponent(this).inject(this)
 
-        viewModel.formsToDisplay.observe(this) { forms -> showFormListDialog(forms) }
+        viewModel.formsToDisplay.observe(this) { forms ->
+            forms?.let {
+                showFormListDialog(it)
```

Comment:
Just realized: this will end up triggering on `onResume` as well as triggering multiple times when the forms are updated via disk sync and db loading etc. We could show the dialog once and set an adapter that's updated with the `LiveData` or create a `DialogFragment` that does something similar. Alternatively, we could just show the forms as a list in `Activity` directly (with a basic `RecyclerView`/`Adapter` combo). I don't think there's any reason to use a dialog here.

Reply:
> Just realized: this will end up triggering on onResume

This is ok because we need to create this dialog again to display it.

> as well as triggering multiple times when the forms are updated via disk sync and db loading

Yes this is not perfect but it would be a very rare case I guess.

> Alternatively, we could just show the forms as a list in Activity directly (with a basic RecyclerView/Adapter combo). I don't think there's any reason to use a dialog here.

I agree but this is something for a separate issue I think

Reply:
> This is ok because we need to create this dialog again to display it.

Right but we're creating a new dialog and calling `show()` multiple times without dismissing the dialog that was previously shown. I tried it out with the debugger, and it looks like we create the dialog a minimum of two times (I'm guessing because of the double `loadFromDatabase` that happens in `BlankFormListViewModel`). However, it seems like this doesn't result in the user having to hit back multiple times to get out of the screen due to the listener passed to `setOnCancelListener` finishing the Activity. There's no visible bug here, but it does feel incorrect and could potentially lead to some confusion or problems in the future so I think we should fix it before moving on.

Reply:
Taking into account that:
- it looks like it's the last thing to fix
- I will be off next 2 weeks 
- the issue was about reworking `BlankFormListActivity` not `AndroidShortcutsActivity`

I think it's the best option to take a step back and keep the old behavior in `AndroidShortcutsActivity` what I did https://github.com/getodk/collect/pull/5077/commits/a04ff5b609abe0b9a94afc0dec3091d39df9ce84

When I'm back I can rework `AndroidShortcutsActivity` separately removing the dialog and displaying a list of forms in a recyclerview.